### PR TITLE
Improve privacy in analytics

### PR DESCRIPTION
**Pull request by yabirgb**
Example response for the API

```
{'system_os': 'Linux', 'system_release': '5.13.19-2-MANJARO', 'system_version': '#1 SMP PREEMPT Sun Sep 19 21:31:53 UTC 2021', 'rotki_version': '1.22.2.dev91+ge0883f9a8', 'country': 'CH', 'city': 'unknown'}
```
In my case I use a VPN and that is why the location resolves to `CH`.

Also fixes a pylint warning

### CODE REVIEW

File: rotkehlchen/usage_analytics.py [hunk 1]
```diff
@@ -22,134 +24,114 @@
...
     country_code: str
-    city: str
 
 
-def query_ipwhoisio() -> Optional[GeolocationData]:
-    """10,000 requests per month per IP
-    https://ipwhois.io/documentation
-    """
+def retrieve_location_data(data_dir: Path) -> Optional[GeolocationData]:
+    # Download a copy of database with maps of ip to country
```
Comment (LefterisJP) [inline rotkehlchen/usage_analytics.py:30]: Add a proper docstring here

File: rotkehlchen/usage_analytics.py [hunk 2]
```diff
@@ -22,134 +24,114 @@
...
+        metadata_query_failed = True
+    except DeserializationError as e:
+        log.debug(f'Failed to deserialize date in metadata information for geoip file. {str(e)}')
+        metadata_query_failed = True
+    except KeyError as e:
+        log.debug(f'Github response for geoip file had missing key {str(e)}')
+        metadata_query_failed = True
+
+    if metadata_query_failed:
+        old_files = glob.glob(str(geoip_dir) + '/*.mmdb')
```
Comment (LefterisJP) [inline rotkehlchen/usage_analytics.py:57]: Use the globbing functionality of Pathlib: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.match

here and below.

File: rotkehlchen/usage_analytics.py [hunk 3]
```diff
@@ -22,134 +24,114 @@
...
-    try:
-        response = requests.get(
-            'http://api.ipstack.com/check?access_key=affd920d6e1008a614900dbc31d52fa6',
-            timeout=LOCATION_DATA_QUERY_TIMEOUT,
-        )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
+        user_ip = requests.get(
```
Comment (LefterisJP) [inline rotkehlchen/usage_analytics.py:85]: We should also fix this. There must be a better way to get the ip of the user than checking an external service.

I know, I know ... you can only reliably find the LAN ip and no non-external solution is perfect. I think we can use miniupnp for this. Which tries to query the user's router. If that fails I think it's fair to simply not use a location in usage analytics.

Example here: https://stackoverflow.com/a/41385033/110395

You should study how miniupnpc works and also do proper error checking. In all errors just don't send location in usage analytics.

File: rotkehlchen/usage_analytics.py [hunk 4]
```diff
@@ -22,134 +24,114 @@
...
+        with maxminddb.open_database(filename) as reader:
+            data = reader.get(user_ip)
+            location = data['country']['iso_code']
+            if location == 'None':
+                return None
+            return GeolocationData(country_code=location)
+    except maxminddb.errors.InvalidDatabaseError as e:
+        os.remove(filename)
+        log.debug(f'Failed to read database {str(e)}')
+    except ValueError as e:
```
Comment (LefterisJP) [inline rotkehlchen/usage_analytics.py:103]: When can this happen?

File: rotkehlchen/usage_analytics.py [hunk 5]
```diff
@@ -22,134 +24,114 @@
...
 
-    location_data = retrieve_location_data()
+    location_data = retrieve_location_data(data_dir)
     if location_data is None:
-        location_data = GeolocationData('unknown', 'unknown')
+        location_data = GeolocationData('unknown')
 
     analytics['country'] = location_data.country_code
-    analytics['city'] = location_data.city
+    analytics['city'] = 'unknown'
```
Comment (LefterisJP) [inline rotkehlchen/usage_analytics.py:121]: write in a comment we don't need this and we deprecate it. Only keeping it here for api compatibility

**Review by LefterisJP — COMMENTED** [review]
some comments

File: rotkehlchen/usage_analytics.py (hunk 4 above)
Comment (yabirgb) [reply rotkehlchen/usage_analytics.py:103]: This happens for example when the provided ip is not valid for the file, for example checking IPv4 agains a IPv6 file

Comment (LefterisJP) [reply rotkehlchen/usage_analytics.py:109]: them raising a simple ValueError is really bad practice on their side. But then at least catch it with a comment explaining why and when this happens.

File: rotkehlchen/tests/unit/test_usage_analytics.py [hunk 6]
```diff
@@ -4,16 +4,15 @@
...
+    analytics = create_usage_analytics(data_dir)
 
     assert 'system_os' in analytics
     assert 'system_release' in analytics
     assert 'system_version' in analytics
     assert analytics['rotki_version'] == get_system_spec()['rotkehlchen']
     if sys.platform != 'darwin':
-        assert analytics['country'] != 'unknown'
-        assert analytics['city'] != 'unknown'
+        assert analytics['city'] == 'unknown'
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/test_usage_analytics.py:15]: Had to remove this from tests since it fails to detect the ip in tests

Comment (LefterisJP) [reply rotkehlchen/tests/unit/test_usage_analytics.py:15]: understandable. We would need to test it in few different networks to make sure it works -- by hand.

File: rotkehlchen/usage_analytics.py [hunk 7]
```diff
@@ -22,134 +24,123 @@
...
+            return None
+        filename = old_files[0]
+    else:
+        filename = geoip_dir / f'geoip-{date}.mmdb'
+
+    if not filename.is_file():
+        # Remove old files
+        files = geoip_dir.glob('*.*')
+        for f in files:
+            os.remove(f)
```
Comment (LefterisJP) [inline rotkehlchen/usage_analytics.py:71]: You use pathlib elsewhere. Use it here too: https://docs.python.org/3/library/pathlib.html#pathlib.Path.unlink

File: rotkehlchen/usage_analytics.py [hunk 8]
```diff
@@ -22,134 +24,123 @@
...
-    # TODO: Add more fallbacks
-    return location_data
+        with maxminddb.open_database(filename) as reader:
+            data = reader.get(user_ip)
+            location = data['country']['iso_code']
+            if location == 'None':
+                return None
+            return GeolocationData(country_code=location)
+    except maxminddb.errors.InvalidDatabaseError as e:
+        os.remove(filename)
```
Comment (LefterisJP) [inline rotkehlchen/usage_analytics.py:107]: use pathlib unlink

File: rotkehlchen/usage_analytics.py [hunk 9]
```diff
@@ -22,134 +24,123 @@
...
-    if response.status_code != HTTPStatus.OK:
+        u = miniupnpc.UPnP()
+        u.discoverdelay = 200
+        u.discover()
+        u.selectigd()
+        user_ip = u.externalipaddress()
+    except Exception as e:  # pylint: disable=broad-except
+        # In my case I was getting a generic exception rised
+        # when failing to detect the ip
+        log.debug(f'Failed to get ip for analytics. {str(e)}')
```
Comment (LefterisJP) [inline rotkehlchen/usage_analytics.py:96]: Don't write things like this in a comment. Tell me that in discord or as a comment in your own PR. 
```suggestion
        # can raise base exception so we catch it
        log.debug(f'Failed to get ip via UPnP for analytics. {str(e)}')
```

File: rotkehlchen/usage_analytics.py [hunk 10]
```diff
@@ -22,134 +24,123 @@
...
     if location_data is None:
-        location_data = GeolocationData('unknown', 'unknown')
+        location_data = GeolocationData('unknown')
 
     analytics['country'] = location_data.country_code
-    analytics['city'] = location_data.city
+    # To improvice privacy of the users we have changed the method of obtaining information
+    # about the country of the user and the city is not information we get anymore. We leave
+    # the field city for compatibility with the API.
+    analytics['city'] = 'unknown'
```
Comment (LefterisJP) [inline rotkehlchen/usage_analytics.py:130]: This is a test. You just write a few words, not an entire paragraph :smile: 

```suggestion
    analytics['city'] = 'unknown'  # deprecated -- we no longer use it
```

**Review by LefterisJP — COMMENTED** [review]
Left you some comments to fix. Works for me as per the screenshot.

File: rotkehlchen/usage_analytics.py (hunk 4 above)
Comment (LefterisJP) [reply rotkehlchen/usage_analytics.py:112]: you did not do this. Anyway I will do it at some other point.

**Review by LefterisJP — APPROVED** [review]
LGTM.

**Pull request merged**